bitcoin/src/error.rs

```diff
@@ -1,7 +1,53 @@
 //! Contains error types and other error handling tools.
 
+use core::fmt;
+
+use bitcoin_internals::write_err;
+
+use crate::consensus::encode;
 pub use crate::parse::ParseIntError;
 
+/// A general error code, other errors should implement conversions to/from this
+/// if appropriate.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum Error {
+    /// Encoding error
+    Encode(encode::Error),
+    /// The header hash is not below the target
+    BlockBadProofOfWork,
+    /// The `target` field of a block header did not match the expected difficulty
+    BlockBadTarget,
+}
```

Comment:
Where is this error even useful? It's not entirely general because it has only three variants and we have many other error types. `validate_pow` doesn't need to decode the block, so `Encode` variant doesn't make sense. It looks like we should have `pow::Error` containing only `BadTarget` and `BadProofOfWork`

Reply:
It used to be our most general error type, before other cleanups :)

It's possible that it's useless now. But we should revisit in another PR.